lib/setup/setupSetup.js

```diff
@@ -272,8 +272,19 @@ function Setup(options) {
         }
     }*/
 
+    /**
+     * Asks the user if he wants to migrate objects if it makes sense and performs migration according to input
+     *
+     * @param {object} newConfig - updated config
+     * @param {object} oldConfig - previous config
+     * @param {object} rl - readline object
+     * @param {function(number=):void} callback - callback function
+     */
     function migrateObjects(newConfig, oldConfig, rl, callback) {
-        if (oldConfig && (oldConfig.states.type !== newConfig.states.type || oldConfig.objects.type !== newConfig.objects.type)) {
+        // allow migration if one of the db types changed or host changed of redis
+        if (oldConfig && ((oldConfig.states.type !== newConfig.states.type || oldConfig.objects.type !== newConfig.objects.type) ||
+            (oldConfig.states.type === 'redis' && oldConfig.states.host !== newConfig.states.host) ||
+            (oldConfig.objects.type === 'redis' && oldConfig.objects.host !== newConfig.objects.host))) {
```

Comment:
I'm not sure about the implications. @Apollon77 ?

Reply:
I must say, I don't understand if the host changed, where we will get the data?

Reply:
The whole PR is basically for this change. I switched my system from one redis server to another one and I was not offered a migration ... Migration should work as before (iobroker backup will contain everything and is then restored on new config ... Seems ok for me but yes needs proper testing ... I assume @foxriver76 tested it :-)